### ISSUE TITLE
Plasma 6 isn't defining pinentry package with mkDefault

### ISSUE THREAD

**Issue by materusPL** [NixOS/nixpkgs#296390]
### Describe the bug
When enabling plasma6, can't use `programs.gnupg.agent.pinentryPackage`

### Steps To Reproduce
1. Enable plasma 6
2. Set up pinentry package

### Expected behavior
Pinentry package is setup to one defined in config

### Additional context
Seems `programs.gnupg.agent.pinentryPackage` is set up in `nixos/modules/services/desktop-managers/plasma6.nix` without using mkDefault or mkForce, so when trying to build system I'm getting
```
       error: The option `programs.gnupg.agent.pinentryPackage' is defined multiple times while it's expected to be unique.

       Definition values:
       - In `/nix/store/bv7ablpcvkf1k5r7ivz6nsz3w604p801-source/configurations/host/materusPC/kde.nix': <derivation pinentry-gnome3-1.2.1>
       - In `/nix/store/9mf8fkjw3vw1gm4s6j5fqlwrv5wf2knf-source/nixos/modules/services/desktop-managers/plasma6.nix': <derivation pinentry-qt-1.2.1>
       Use `lib.mkForce value` or `lib.mkDefault value` to change the priority on any of these definitions.

```

### Notify maintainers

@fpletz @SuperSandro2000  // Not sure of module maintainers so adding authors of commit

### Metadata
```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.9-zen1, NixOS, 24.05 (Uakari), 24.05.20240309.3030f18`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.20.5`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/nra828scc8qs92b9pxra5csqzffb6hpl-source`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Emantor** [comment]
The fix is presumably #295891.

**Comment by SuperSandro2000** [comment]
> When enabling plasma6, can't use programs.gnupg.agent.pinentryPackage

That isn't true, you can just use lib.mkForce to overwrite the package.
